Cpp_test/lib/ipc-shm-send.cpp

```diff
@@ -0,0 +1,98 @@
+#include "ipc-shm-send.h"
+
+IPCShmSend::~IPCShmSend()
+{
+	shm_unlink(this->info.method_name);
+}
+
+void IPCShmSend::init()
+{
+	this->file_handler.open_file();
+	this->open_shm();
+	this->set_shm_size();
+	this->map_shm();
+	this->init_mutex();
+	this->init_cond();
+}
+
+void IPCShmSend::transfer()
+{
+	long read_bytes{0};
+	unsigned long long file_size{this->file_handler.get_file_size()};
+	if (file_size == 0)
+	{
+		throw std::runtime_error("ERROR: File size = 0.");
+	}
+
+	unsigned long long total_sent_bytes{0};
+	std::cout << "Waiting for receiver..." << std::endl;
+	while (!this->is_end)
+	{
+		lock_mutex();
+	
+		if (this->shm_ptr->is_read)
```

Comment:
It is easier for me to check transfer between receiver and sender

Reply:
I mean, this is mandatory, it was just for your information ( and the limit of condvars)

Reply:
Because here I only want to use condvars in receiver, sender will take responsibility for making signal then the receiver will wait and then send back so I need to read the is_read all the time

Reply:
Yes yes I know, it was just for your information :)

Reply:
I think that the issue here is that the sender unlocks the mutex, broadcasts the signal, and ... lock against the mutex (the receiver doesn't have time to lock the mutex).

Reply:
Maybe it works like that, however, I see the program still works so well so I think it is not necessary to change

Reply:
As I said, it was for your information and to give you some elements if you want to improve your program. If you don't want... it's your business